src/evict/evict_lru.c

```diff
@@ -24,40 +24,48 @@ static int  __evict_walk_file(
 	(S2C(s)->evict_threads.current_threads > 1)
 
 /*
- * __evict_lock_dhandle --
- *	Try to get the dhandle lock, with yield and sleep back off.
+ * __evict_lock_handle_list --
+ *	Try to get the handle list lock, with yield and sleep back off.
  *	Keep timing statistics overall.
  */
 static int
-__evict_lock_dhandle(WT_SESSION_IMPL *session)
+__evict_lock_handle_list(WT_SESSION_IMPL *session)
 {
 	struct timespec enter, leave;
 	WT_CACHE *cache;
 	WT_CONNECTION_IMPL *conn;
 	WT_DECL_RET;
-	WT_SPINLOCK *dh_lock;
-	int64_t **stats;
+	WT_RWLOCK *dh_lock;
 	u_int spins;
 	bool dh_stats;
 
 	conn = S2C(session);
 	cache = conn->cache;
 	dh_lock = &conn->dhandle_lock;
-	stats = (int64_t **)conn->stats;
+	/*
+	 * We want to know if a lot of time is spent waiting for this readlock
+	 * but don't (yet) have stats for read/write locks.
+	 */
+#if 0 /* XXX */
```

Comment:
As a concrete suggestion, I think we should remove the three handle-list lock stats:
```

    LockStat('lock_handle_list_count', 'handle-list lock acquisitions'),
    LockStat('lock_handle_list_wait_application', 'handle-list lock application thread wait time (usecs)'),
    LockStat('lock_handle_list_wait_internal', 'handle-list lock internal thread wait time (usecs)'),
```

And add a new stat, ``Eviction server time spent waiting for handle list lock (usecs)``, and re-enable this code.